File: test_dir/cpp/src/strings/exceptions.cpp
#include "strings\exceptions.h"

void string_exceptions_fn() {
    print(PyStr("STRING EXCEPTIONS RESULTS:"));
    PyStr s = PyStr("test");
    try {
        s.index(PyStr("a"));
    } catch (const PyppValueError &pypp_e) {
        std::string e = pypp_e.what();
        print(PyStr("value error: ") + to_pystr(e));
    }
    try {
        s.rindex(PyStr("a"));
    } catch (const PyppValueError &pypp_e) {
        std::string e = pypp_e.what();
        print(PyStr("value error: ") + to_pystr(e));
    }
    try {
        s[9];
    } catch (const PyppIndexError &pypp_e) {
        std::string e = pypp_e.what();
        print(PyStr("index error: ") + to_pystr(e));
    }
    try {
        s[-9];
    } catch (const PyppIndexError &pypp_e) {
        std::string e = pypp_e.what();
        print(PyStr("index error: ") + to_pystr(e));
    }
}

File: test_dir/python/src/strings/exceptions.py
def string_exceptions_fn():
    print("STRING EXCEPTIONS RESULTS:")
    s: str = "test"
    try:
        s.index("a")
    except ValueError as e:
        print("value error: " + str(e))
    try:
        s.rindex("a")
    except ValueError as e:
        print("value error: " + str(e))
    try:
        s[9]
    except IndexError as e:
        print("index error: " + str(e))
    try:
        s[-9]
    except IndexError as e:
        print("index error: " + str(e))
